### PR TITLE
parse json_table syntax

### PR DESCRIPTION
**Pull request by JCOR11599**
TODO:
 - `NESTED`
 - `FOR ORDINALITY`
 - `ON EMPTY`
 - `ON ERROR`

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -4409,11 +4421,17 @@ partition_list:
   }
 
 table_function:
-  ID openb argument_expression_list_opt closeb
+  // TODO: handle NESTED
+  JSON_TABLE openb STRING ',' STRING COLUMNS openb table_column_list closeb closeb AS table_alias
```
[the remark this answers is not in the document]
Comment (zachmu) [reply go/vt/sqlparser/sql.y:4425]: table_column_list isn't correct here, you need to support a different syntax for this function:

```
column:
    name FOR ORDINALITY
    |  name type PATH string path [on_empty] [on_error]
    |  name type EXISTS PATH string path
    |  NESTED [PATH] path COLUMNS (column_list)
```

**Review by zachmu — COMMENTED** [review]
Good start but there's a lot missing here.

Also needs parser tests

File: go/vt/sqlparser/sql.y (hunk 1 above)
[the remark this answers is not in the document]
Comment (JCOR11599) [reply go/vt/sqlparser/sql.y:4425]: Oddly, the `alias` is required for `json_table`
https://dev.mysql.com/doc/refman/8.0/en/json-table-functions.html#function_json-table

File: go/vt/sqlparser/parse_test.go [hunk 2]
```diff
@@ -6409,6 +6409,76 @@ func TestKeywordsIncorrectlyDontParse(t *testing.T) {
...
+FROM
+	JSON_TABLE(
+		'[{"a":1},{"a":2}]',
+		"$[*]" COLUMNS(
+			x varchar(100) path "$.a"
+		)
+	) as tt;`,
+			output: `select * from JSON_TABLE('[{\"a\":1},{\"a\":2}]', "$[*]" COLUMNS(
+	x varchar(100) path "$.a"
+)) as tt`},
```
Comment (zachmu) [inline go/vt/sqlparser/parse_test.go:6426]: Make sure you have at least one test that omits the optional `as` keyword

File: go/vt/sqlparser/sql.y [hunk 3]
```diff
@@ -4468,6 +4471,7 @@ table_references:
 table_reference:
   table_factor
 | join_table
+| json_table
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:4474]: This should be a table_factor, not a table_reference

Comment (zachmu) [reply go/vt/sqlparser/sql.y:4474]: Make sure you have parse tests for using json_table in joins, in unions, in subqueries, etc.

File: go/vt/sqlparser/sql.y [hunk 4]
```diff
@@ -4711,6 +4716,83 @@ natural_join:
     }
   }
 
+json_table:
+  JSON_TABLE openb value_expression ',' STRING COLUMNS openb json_table_column_list closeb closeb AS table_alias
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:4720]: AS needs to be optional here

**Review by zachmu — APPROVED** [review]
LGTM, just a few comments and tests to add

**Pull request merged**